akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -507,15 +507,43 @@ object HttpEntity {
       withSizeLimit(SizeLimit.Disabled)
 
     override def transformDataBytes(transformer: Flow[ByteString, ByteString, Any]): HttpEntity.Chunked = {
-      val newData =
-        chunks.map {
-          case Chunk(data, "")    => data
-          case LastChunk("", Nil) => ByteString.empty
-          case _ =>
-            throw new IllegalArgumentException("Chunked.transformDataBytes not allowed for chunks with metadata")
-        } via transformer
-
-      HttpEntity.Chunked.fromData(contentType, newData)
+      import akka.stream.scaladsl.GraphDSL.Implicits._
+
+      val buf = scala.collection.mutable.ArrayBuffer.empty[Throwable]
```

Comment:
Could you explain why this is needed? The problem with introducing mutable state here, is that an otherwise reusable entity becomes a single use entity that will fail in weird ways. So, any solution cannot depend on mutable state like this.

Reply:
It will hold at most one throwable, using that is the leftover of some experimenting, will remove that

Reply:
I see what the problem was. Without that solution, potential errors might skip the user-supplied transformer because they race through the second branch of the bcast/concat. I've posted a simple solution in the alternative snippet.